Userland/Libraries/LibVideo/PlaybackManager.cpp

```diff
@@ -488,20 +536,31 @@ class PlaybackManager::PausedStateHandler : public PlaybackManager::PlaybackStat
     PlaybackState get_state() const override { return PlaybackState::Paused; }
 };
 
+// FIXME: This is a placeholder variable that could be scaled based on how long each frame decode takes to
+//        avoid triggering the timer to check the queue constantly. However, doing so may reduce the speed
+//        of seeking due to the decode thread having to wait for a signal to continue decoding.
+constexpr int buffering_or_seeking_decode_wait_time = 1;
+
 class PlaybackManager::BufferingStateHandler : public PlaybackManager::ResumingStateHandler {
     using PlaybackManager::ResumingStateHandler::ResumingStateHandler;
 
     ErrorOr<void> on_enter() override
     {
-        manager().m_decode_timer->start(0);
+        manager().m_present_timer->start(buffering_or_seeking_decode_wait_time);
         return {};
     }
 
     StringView name() override { return "Buffering"sv; }
 
-    ErrorOr<void> on_buffer_filled() override
+    ErrorOr<void> on_timer_callback() override
     {
-        return assume_next_state();
+        auto buffer_is_full = manager().m_buffer_is_full.load();
+        dbgln_if(PLAYBACK_MANAGER_DEBUG, "Buffering timer callback has been called. Buffer is {}.", buffer_is_full ? "full, exiting"sv : "not full, waiting"sv);
+        if (buffer_is_full)
+            return assume_next_state();
+
+        manager().m_present_timer->start(buffering_or_seeking_decode_wait_time);
```

Comment:
reaching into friend classes to call methods on their private variables feels very awkward imo. I would rather see a `start_presentation_timer` public and/or Badged API added to the manager object.

Reply:
To me, given that these state handlers aren't (and perhaps can't be) designed to be extensible, and are meant to be entirely internal state that nothing outside is should see or interact with, it seems like adding those boilerplate private getters/etc that would be required would just add a lot of unnecessary bloat to the file. I'll leave the decision up to you though, let me know if you want me to implement it that way.

Reply:
It may be worth adding at least a private `start_presentation_timer()` function as you mentioned, though, since the lines for starting the timer do look slightly ugly.

Reply:
Forgot to mention, but I did make a private function to set the timer and renamed it to better suit its purpose.